### PR TITLE
vlc: Add chromecast support; libmicrodns: Init at 0.0.10

### LINKED ISSUE #58365: vlc: Add chromecast support
The current vlc derivation does not include chromecast support.

I believe we need to add the flags `--enable-sout --enable-chromecast` during the configuration step to enable it by default, which I think is a reasonable thing to do.

I tried to do the minimal change myself, but looks like we need some new dependencies that don't exist in nixpkg yet:

```diff
diff --git a/pkgs/applications/video/vlc/default.nix b/pkgs/applications/video/vlc/default.nix
index cd3999e38c0..777e302e8e5 100644
--- a/pkgs/applications/video/vlc/default.nix
+++ b/pkgs/applications/video/vlc/default.nix
@@ -76,6 +76,8 @@ stdenv.mkDerivation rec {
   # "--enable-foo" flags here
   configureFlags = [
     "--with-kde-solid=$out/share/apps/solid/actions"
+    "--enable-sout"
+    "--enable-chromecast"
   ] ++ optional onlyLibVLC "--disable-vlc";
```

```
$ nix-env -i vlc -f $PWD
...
checking for protoc... no
checking for protobuf-lite >= 2.5.0... no
configure: error: Library protobuf-lite >= 2.5.0 needed for chromecast was not found
builder for '/nix/store/n8ky60h4n2r67agrgag5m5znijk2df59-vlc-3.0.6.drv' failed with exit code 1
```

I'm still fairly new so it will take some learning to figure out what needs to be done to package protobuf-lite (not sure I'll get around to it anytime soon).

If anyone else is into making these changes, it would be much appreciated. Or please let me know if I'm off-track here and there's a better way to do this.

### PR DESCRIPTION
**Pull request by shazow**
###### Motivation for this change

VLC normally ships with Chromecast support enabled by default, in the official binaries as well as in other distros (arch, ubuntu, possibly others).

Chromecast support for VLC has been requested on several occasions (https://github.com/NixOS/nixpkgs/pull/35124#issuecomment-385791097, #58365 from me, and [a bunch of times on IRC](https://www.google.ca/search?q="nixos"+"chromecast"+"vlc"+"irc+logs")). 

This PR adds libmicrodns which is a library written by the VLC team that is required for their Chromecast streaming to function.

It also adds Chromecast support to the VLC build enabled by default, but can be overridden.

Fixes #58365.

This is my first nixpkgs contribution, please be gentle. :blush: 

Some notes/questions:
- I'm not sure if this is a strict requirement, but enabling hardware acceleration is important for transcoding especially: https://nixos.wiki/wiki/Accelerated_Video_Playback - I needed to add `intel-media-driver ` to get `libva info: va_openDriver() returns -1` errors to stop. Not sure what's a more generalized solution. Is there a good place to document this? (Otherwise VLC spits out a bunch of errors, might fall back or not)
- Port 8010 must be open (streaming works by creating a temporary HTTP server on port 8010 which streams the transcoded video, then points the Chromecast to open that local URL). Any suggestions for documenting/enforcing this?

###### Things done

- [ ] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nix-review --run "nix-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [x] Assured whether relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

---

### CODE REVIEW

File: maintainers/maintainer-list.nix [hunk 1]
```diff
@@ -5391,4 +5391,9 @@
     github = "minijackson";
     name = "Rémi Nicole";
   };
+  shazow = {
```
Comment (shazow) [inline maintainers/maintainer-list.nix]: The last several entries in the maintainer list aren't alphabetic. Should I reorder the whole thing, or are we doing appending now?

**Comment by Infinisil** [comment]
Also related: https://github.com/NixOS/nixpkgs/pull/53184

**Comment by shazow** [comment]
@Infinisil Good call, this PR should obsolete #35184, and it also makes scanning/autodetecting work properly (needed libmicrodns).

**Comment by flokli** [comment]
> Port 8010 must be open (streaming works by creating a temporary HTTP server on port 8010 which streams the transcoded video, then points the Chromecast to open that local URL). Any suggestions for documenting/enforcing this?

I'd drop a short comment somewhere in the vlc derivation, and add a longer `<para>` inside `nixos/doc/manual/release-notes/rl-1909.xml`, "Other Notable Changes", noting VLC gained Chromecast support, suggesting to set `networking.firewall.allowedTCPPorts = [8010]`.

Does VLC document that port somehwere obvious inside their documentation?

File: maintainers/maintainer-list.nix (hunk 1 above)
Comment (flokli) [reply maintainers/maintainer-list.nix]: I'd just order it in a commit before the one adding yourself.

**Comment by flokli** [comment]
@shazow you might want to add `protoc` to both `buildInputs` and `nativeBuildInputs`, see https://github.com/NixOS/nixpkgs/pull/53184#discussion_r244846577.

File: pkgs/development/libraries/libmicrodns/default.nix [hunk 2]
```diff
@@ -0,0 +1,30 @@
+{ stdenv
+, fetchFromGitHub
+, autoreconfHook
+, pkgconfig
+}:
+
+stdenv.mkDerivation rec {
+  version = "0.0.10";
+  name = "libmicrodns-${version}";
```
Comment (ryantm) [inline pkgs/development/libraries/libmicrodns/default.nix]:
```suggestion
  pname = "libmicrodns";
```

File: pkgs/development/libraries/libmicrodns/default.nix [hunk 3]
```diff
@@ -0,0 +1,30 @@
...
+
+  src = fetchFromGitHub {
+    owner = "videolabs";
+    repo = "libmicrodns";
```
Comment (ryantm) [inline pkgs/development/libraries/libmicrodns/default.nix]:
```suggestion
    repo = pname;
```

**Review by ryantm — CHANGES_REQUESTED** [review]

File: maintainers/maintainer-list.nix (hunk 1 above)
Comment (ryantm) [reply maintainers/maintainer-list.nix]: I think the ordering of the maintainer list is not related to this PR, so I recommend not worrying about it. I'm worried the reordering will cause a bunch of merge conflicts in new PRs too.

**Comment by shazow** [comment]
Pushed squashed changes.

What's the easiest way to verify that the release notes are rendered properly?

**Comment by flokli** [comment]
@shazow from your checkout:

```
⇒  nix-build nixos/release.nix -A manual.x86_64-linux
⇒  cd result
⇒  python3 -m http.server
```

Access `http://localhost:8000/share/doc/nixos/release-notes.html#sec-release-19.09-notable-changes` in your browser

**Comment by shazow** [comment]
@flokli Ah thank you!

Alright, pushed some more minor edits, should be good for another look:

image

File: pkgs/applications/video/vlc/default.nix [hunk 4]
```diff
@@ -13,8 +13,13 @@
 , jackSupport ? false
 , fetchpatch
 , removeReferencesTo
+, chromecastSupport ? true, protobuf, libmicrodns
 }:
 
+# chromecastSupport requires TCP port 8010 to be open for it to work.
+# If your firewall is enabled, make sure to have something like:
+#   networking.firewall.allowedTCPPorts = [ 8010 ]
```
Comment (flokli) [inline pkgs/applications/video/vlc/default.nix]:
```suggestion
#   networking.firewall.allowedTCPPorts = [ 8010 ];
```

**Review by flokli — APPROVED** [review]
minor nitpick, apart from that, LGTM!

**Comment by ryantm** [comment]
@grahamcofborg build vlc

File: pkgs/applications/video/vlc/default.nix (hunk 4 above)
Comment (shazow) [reply pkgs/applications/video/vlc/default.nix]: Squashed into the commit, thanks!

**Pull request merged**